include/crab/domains/region_domain.hpp

```diff
@@ -561,6 +567,30 @@ class region_domain final
   base_abstract_domain_t m_base_dom;
   /** End base domain **/
 
+  /** Begin MRU domain **/
+  // The cache abstract domain that capture invariants for MRU object
+  base_abstract_domain_t m_cache_dom;
+  using base_variable_domain_t = variable_domain<base_variable_t>;
+  // left: the base address for cache
+  // right: a region variable to get equivalence class
+  using cache_variable_opt_t =
+      boost::optional<std::pair<base_variable_domain_t, variable_t>>;
+  cache_variable_opt_t m_cache_vars;
+  // key: a reference variable
+  // value: a base variable in the base domain that represents the base address
+  // of the MRU
+  using base_var_map_t =
+      ikos::separate_domain<variable_t, base_variable_domain_t>;
```

Comment:
The current environment cannot resolve the following code:
```
p := make_ref(…)
q := make_ref(….)
if (…) {
r := gep_ref(p, …);
} else {
r := gep_ref(q, …)
}
store_ref(r,...)
```
The reason is the domain only maps from one reference variable into a single base address.

Reply:
yes, we can have directly a set of variables (`discrete_domains.hpp`) to keep track of multiple possible base addresses. But can we do something useful when the set is not a singleton?